### PR TITLE
Add support for initial check status

### PR DESCRIPTION
**Pull request by hoffoo**
accidentally closed the pr, here is a proper commit for initial check status

old thread: #1573

### CODE REVIEW

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -1653,6 +1655,17 @@ func (sc *ServiceCheck) validate() error {
...
 	}
 
+	switch sc.InitialStatus {
+	case "":
+	case api.HealthUnknown:
+	case api.HealthPassing:
+	case api.HealthWarning:
+	case api.HealthCritical:
+	default:
+		return fmt.Errorf(`invalid initial check state (%s), must be one of "%s", "%s", "%s", or "%s"`, sc.InitialStatus, api.HealthUnknown, api.HealthPassing, api.HealthWarning, api.HealthCritical)
```
Comment (diptanu) [inline nomad/structs/structs.go]: `invalid initial check state (%s), must be one of %q, %q, %q, %q or empty`

**Comment by diptanu** [comment]
@hoffoo Thanks! Left one small comment.

File: jobspec/parse_test.go [hunk 2]
```diff
@@ -525,3 +525,17 @@ func TestIncorrectKey(t *testing.T) {
...
+
+func TestServiceCheckInitialStatus(t *testing.T) {
+
+	path, err := filepath.Abs(filepath.Join("./test-fixtures", "service-check-initial-status.hcl"))
+	if err != nil {
+		t.Fatalf("Can't get absolute path for file: %s", err)
+	}
+
+	_, err = ParseFile(path)
+
```
Comment (diptanu) [inline jobspec/parse_test.go]: We should also check that parsing the file resulted in the check that we are expecting by using `reflect.DeepEquals`

Comment (hoffoo) [inline jobspec/parse_test.go]: okay! it turns out it was actually broken without the mapstructure tag :x

**Comment by diptanu** [comment]
@hoffoo Thanks! I am merging this one.

**Pull request merged**